starter-code/memory.js

```diff
@@ -2,61 +2,138 @@
-  		{ name: "green arrow",     img: "green-arrow.jpg" },
-  		{ name: "green lantern",   img: "green-lantern.jpg" },
-  		{ name: "ironman",         img: "ironman.jpg" },
-  		{ name: "spiderman",       img: "spiderman.jpg" },
-  		{ name: "superman",        img: "superman.jpg" },
-  		{ name: "the avengers",    img: "the-avengers.jpg" },
-  		{ name: "thor",            img: "thor.jpg" },
-  	];
+    this.cards = [
+        { name: "aquaman", img: "aquaman.jpg" },
+        { name: "batman", img: "batman.jpg" },
+        { name: "captain america", img: "captain-america.jpg" },
+        { name: "fantastic four", img: "fantastic-four.jpg" },
+        { name: "flash", img: "flash.jpg" },
+        { name: "green arrow", img: "green-arrow.jpg" },
+        { name: "green lantern", img: "green-lantern.jpg" },
+        { name: "ironman", img: "ironman.jpg" },
+        { name: "spiderman", img: "spiderman.jpg" },
+        { name: "superman", img: "superman.jpg" },
+        { name: "the avengers", img: "the-avengers.jpg" },
+        { name: "thor", img: "thor.jpg" },
+        { name: "aquaman", img: "aquaman.jpg" },
+        { name: "batman", img: "batman.jpg" },
+        { name: "captain america", img: "captain-america.jpg" },
+        { name: "fantastic four", img: "fantastic-four.jpg" },
+        { name: "flash", img: "flash.jpg" },
+        { name: "green arrow", img: "green-arrow.jpg" },
+        { name: "green lantern", img: "green-lantern.jpg" },
+        { name: "ironman", img: "ironman.jpg" },
+        { name: "spiderman", img: "spiderman.jpg" },
+        { name: "superman", img: "superman.jpg" },
+        { name: "the avengers", img: "the-avengers.jpg" },
+        { name: "thor", img: "thor.jpg" },
+    ];
     this.selectedCards = [];
     this.pairsClicked = 0;
     this.correctPairs = 0;
+    this.correctPairsArray = []; // So we can access the already correct pairs later
+    this._shuffleCards() // Shuffle the board on each new instance (prototype on line 42)
 };
 
 // //******************************************************************
+// // Game Functions (prototypes)
+// //******************************************************************
+
+MemoryGame.prototype._shuffleCards = function() {
+    this.cards = _.shuffle(this.cards)
+}
+
+MemoryGame.prototype.selectCard = function(card) {
+
+    this.selectedCards.push($(card).attr('name')) // Push card on the selected cards array for later comparison
+
+    this.correctPairsArray.push(card) // Push card on the correct cards array so we can access them later
+
+    if (this.selectedCards.length === 2) { // A pair is on screen
+
+        if (this.selectedCards[0] === this.selectedCards[1]) { // Cards are the same
+            this.correctPairs++; // Increment correct count
+            this.pairsClicked++; // Increment clicked count
+
+            // Memory Board actions
+
+            for (i = 0; i < this.correctPairsArray.length; i++) { // Loop to leave correct pairs on front side
```

Comment:
Good approach using logic to hold an array of correct pairs. However,  notice that the loop is causing a bit of "jumpy" behavior on the page when the game is played. I'm not sure why, but it could be related to your "workaround" :)